docs/WritingTests.asciidoc

```diff
@@ -752,17 +768,169 @@ Sending key presses over VNC is very slow so for tests which send a lot of
 text commands it is much faster to use a serial port for both sending and
 receiving TTY commands.
 
+Communicating entirely using text also means that you no longer have to worry
+about your needles being invalidated due to a font change or similar. It is
+also much cheaper to transfer text and test it against regular expressions
+than encode images from a VNC feed and test them against sample images
+(needles).
+
+On the other hand you can no longer use +assert_screen+ or take a screen shot
+because the text is never rendered as an image. A lot of programs will also
+send ANSI escape sequences which will appear as raw text to the test script
+instead of being interpreted by a terminal emulator which then renders the
+text.
+
+However the advantages of using the serial terminal usually outweigh the
```

Comment:
Very discussable statement , would better remove it. It's up to openQA consumer to decide which way is better